### DISCUSSION TITLE
Script ideas / improvements 💪

### DISCUSSION THREAD

**Discussion by Mazimus332** [fedorovvl/tso_client#109] [Ideas]
💡 Sound manager: If possible, a script to manage game sounds, such as deactivating explorers/geologist sending/arrival sounds.

💡 Trades posted on Trade Office historic: I think that I read that auto-posting trades would conflict game rules, so I come up with this workaround that I'm sure doesn't conflict them, and has the same idea and purpose. It would be nice if this trade Historic was extensive, meaning that years later we could see what we posted or bought a while ago on different events, which could also be mentioned like "Halloween Event 2022 6/6/2006 | Offered 6 pumpkins - Requested 6.666.666 gold", also for those who aren't interested on seeing their extensive historic, or want to Clear the Data of the screen, a "Save and Clear Data" would be nice.

⬆️ Explorer/Geologist save/load separate directories: These directories are the same (as we can see in settings.json), no matter which template you save, geologist or explorer, both will path the same directory, and it would be cool if it didn't, just as it happens with Generals userscript, which has its individual directory.
Another suggestion, mostly for Geologist, would be to hide permanently unused Geologists (but resetable), I have tons of Geologists that I don't use and appear there every time that I want to send the good ones.

⬆️ Guild script: Adding the guild shield png on the left side of the name of the guild and its acronyms, add guild CSV default improvement with generally useful data such as: 
![Screenshot_4](https://user-images.githubusercontent.com/116684854/200079882-0be03ff9-7230-4728-a4b8-228bb5e93c93.png)

⬆️ Buff manager: It was already requested that would be cool to be able to switch between popular long-lasting buffs like smoked fish and grilled steak within the UI, it would also be good something like Aunt Irma's Feast switched with Employee of the month badge or any +200% 6-16h buff (for gold mines, copper, iron or any use). 
I know we can edit the text file, but wouldn't it be wonderful if it was "automatic"... 
Also, talking about mines, we have many different versions of mines buildings, so it throws "name mismatch" red background when we place a Special or Industrial Mine instead of a normal one. Something like Special Gold Mine = Industrial Gold Mine = Gold Mine, to avoid "name mismatch" (different building) red background.
Extra: Zone buffs like Farming & Hunting guide can't be registered with buff manager, would be great if got added.

Some of my ideas and requests are too perfectionists, I know that maybe not everything is possible, but I wanted to share my ideas, and you decide what is more convenient dedicating time on :)
Thanks to everyone for the scripts and everything that you are doing for the community [❤️](https://emojipedia.org/red-heart/). If I have more ideas, I'll post later.

**Comment by Urvdmih** [comment]
Для менеджера бафов неплохо было бы еще перенести кнопку "Записать еще" вверх или на тулбар вниз, где уже есть другие кнопки. А то каждый раз приходится портянку бафов скролить вниз, что неудобно.
